### DISCUSSION TITLE
Parse Issue (Xcode): Module 'connectivity_plus' not found

### DISCUSSION THREAD

**Discussion by mohamed-gearbox** [codemagic-ci-cd/codemagic-docs#2053] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. The Flutter app runs correctly on my android emulator and a published bundle also runs correctly when deployed to Google Play.

2. I'm using codemagic to build my flutter ipa and be able to deploy on Apple test flight so I can test my app on iPhone (I can't build locally as I'm not an experienced iOS developer, and don't have macOS device to build on - hence codemagic).

3. The following is what I have in codemagic.yaml file:

workflows:
  ios-workflow:
    name: iOS Workflow
    max_build_duration: 120
    instance_type: mac_mini_m1
    integrations:
      app_store_connect: XXX Mobile App
    environment:
      flutter: stable
      xcode: latest
      vars:
        APP_STORE_APPLE_ID: XXX
      ios_signing:
        distribution_type: development
        bundle_identifier: xxx.xxx.xxx
    scripts:
      - name: Set up code signing settings on Xcode project
        script: | 
          xcode-project use-profiles
      - name: Get Flutter packages
        script: | 
          flutter packages pub get
      - name: Install pods
        script: | 
          find . -name "Podfile" -execdir pod install \;
      - name: Flutter build ipa
        script: | 
          flutter build ipa --release \
            --build-name=1.0.0 \
            --build-number=$(($(app-store-connect get-latest-app-store-build-number "$APP_STORE_APPLE_ID") + 1)) \
            --export-options-plist=/Users/builder/export_options.plist
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - flutter_drive.log
    publishing:
      email:
        recipients:
          - test@example.com
        notify:
          success: true
          failure: true
      app_store_connect:
        auth: integration

        # Configuration related to TestFlight (optional)
        # Note: This action is performed during post-processing.
        submit_to_testflight: true
        
        # Configuration related to App Store (optional)
        # Note: This action is performed during post-processing.
        submit_to_app_store: false

4. The build process fails at the "Flutter build ipa" step, the following is the output of the build process at this step:

Resolving dependencies...
  _fe_analyzer_shared 61.0.0 (64.0.0 available)
  _flutterfire_internals 1.3.1 (1.3.5 available)
  analyzer 5.13.0 (6.2.0 available)
  args 2.4.1 (2.4.2 available)
  build 2.3.1 (2.4.1 available)
  build_daemon 3.1.1 (4.0.0 available)
  build_resolvers 2.2.0 (2.3.2 available)
  build_runner 2.3.3 (2.4.6 available)
  build_runner_core 7.2.2 (7.2.10 available)
  built_value 8.5.0 (8.6.2 available)
  checked_yaml 2.0.1 (2.0.3 available)
  code_builder 4.4.0 (4.6.0 available)
  collection 1.17.2 (1.18.0 available)
  cross_file 0.3.3+4 (0.3.3+5 available)
  cupertino_icons 1.0.5 (1.0.6 available)
  dart_openai 1.9.91 (4.0.0 available)
  dart_style 2.3.1 (2.3.2 available)
  dio 5.1.2 (5.3.2 available)
  elegant_notification 1.10.0 (1.11.1 available)
  fetch_client 1.0.1 (1.0.2 available)
  ffi 2.0.2 (2.1.0 available)
  file 6.1.4 (7.0.0 available)
  file_selector_linux 0.9.1+3 (0.9.2+1 available)
  file_selector_macos 0.9.2 (0.9.3+2 available)
  file_selector_platform_interface 2.5.1 (2.6.1 available)
  file_selector_windows 0.9.2 (0.9.3+1 available)
  firebase_auth 4.5.0 (4.9.0 available)
  firebase_auth_platform_interface 6.15.1 (6.18.0 available)
  firebase_auth_web 5.5.1 (5.8.0 available)
  firebase_core 2.13.0 (2.15.1 available)
  firebase_core_web 2.5.0 (2.7.0 available)
  flutter_cache_manager 3.3.0 (3.3.1 available)
  flutter_lints 2.0.1 (2.0.3 available)
  flutter_plugin_android_lifecycle 2.0.15 (2.0.16 available)
  flutter_secure_storage 6.1.0 (9.0.0 available)
  flutter_secure_storage_linux 1.1.3 (1.2.0 available)
  flutter_secure_storage_macos 1.1.2 (3.0.1 available)
  flutter_secure_storage_platform_interface 1.0.1 (1.0.2 available)
  flutter_secure_storage_web 1.1.1 (1.1.2 available)
  flutter_secure_storage_windows 1.1.3 (3.0.0 available)
  glob 2.1.1 (2.1.2 available)
  graphs 2.2.0 (2.3.1 available)
  http 0.13.6 (1.1.0 available)
  image_picker 0.8.9 (1.0.4 available)
  image_picker_android 0.8.7 (0.8.8 available)
  image_picker_for_web 2.2.0 (3.0.1 available)
  image_picker_ios 0.8.8 (0.8.8+2 available)
  image_picker_linux 0.2.1 (0.2.1+1 available)
  image_picker_macos 0.2.1 (0.2.1+1 available)
  image_picker_platform_interface 2.8.0 (2.9.1 available)
  image_picker_windows 0.2.1 (0.2.1+1 available)
  intl 0.17.0 (0.18.1 available)
  js 0.6.5 (0.6.7 available)
  json_serializable 6.6.2 (6.7.1 available)
  lints 2.0.1 (2.1.1 available)
  logging 1.1.1 (1.2.0 available)
  material_color_utilities 0.5.0 (0.8.0 available)
  meta 1.9.1 (1.10.0 available)
  path_provider 2.0.15 (2.1.1 available)
  path_provider_android 2.0.27 (2.2.0 available)
  path_provider_foundation 2.2.3 (2.3.1 available)
  path_provider_linux 2.1.11 (2.2.1 available)
  path_provider_platform_interface 2.0.6 (2.1.1 available)
  path_provider_windows 2.1.7 (2.2.1 available)
  permission_handler 10.4.3 (11.0.0 available)
  permission_handler_android 10.3.2 (11.0.1 available)
  permission_handler_platform_interface 3.11.3 (3.11.5 available)
  petitparser 5.1.0 (6.0.1 available)
  platform 3.1.0 (3.1.2 available)
  plugin_platform_interface 2.1.4 (2.1.6 available)
  process 4.2.4 (5.0.0 available)
  pubspec_parse 1.1.0 (1.2.3 available)
  retrofit_generator 5.0.0+1 (7.0.8 available)
  shimmer 2.0.0 (3.0.0 available)
  source_gen 1.3.2 (1.4.0 available)
  source_helper 1.3.3 (1.3.4 available)
  sqflite 2.2.8+4 (2.3.0 available)
  sqflite_common 2.4.5+1 (2.5.0 available)
  stack_trace 1.11.0 (1.11.1 available)
  stream_channel 2.1.1 (2.1.2 available)
  syncfusion_flutter_charts 20.4.54 (22.2.12 available)
  syncfusion_flutter_core 20.4.54 (22.2.12 available)
  test_api 0.6.0 (0.6.1 available)
  timing 1.0.0 (1.0.1 available)
  tuple 2.0.1 (2.0.2 available)
  typed_data 1.3.1 (1.3.2 available)
  url_launcher 6.1.11 (6.1.14 available)
  url_launcher_android 6.0.35 (6.1.0 available)
  url_launcher_ios 6.1.4 (6.1.5 available)
  url_launcher_linux 3.0.5 (3.0.6 available)
  url_launcher_macos 3.0.5 (3.0.7 available)
  url_launcher_platform_interface 2.1.2 (2.1.5 available)
  url_launcher_web 2.0.17 (2.0.20 available)
  url_launcher_windows 3.0.6 (3.0.8 available)
  uuid 3.0.7 (4.0.0 available)
  watcher 1.0.2 (1.1.0 available)
  win32 4.1.4 (5.0.7 available)
  xdg_directories 1.0.0 (1.0.3 available)
  xml 6.2.2 (6.4.2 available)
Got dependencies!

Archiving xxx.xxx.xxx...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: XXXXXXXXXX
Running pod install...                                             646ms
Running Xcode build...                                          
Xcode archive done.                                         48.2s
Failed to build iOS app
Parse Issue (Xcode): Module 'connectivity_plus' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.

Build failed :|
Step 7 script `Flutter build ipa` exited with status code 1

I believe the solution is to install the 'connectivity_plus' module on the Xcode running on the mac_mini_m1 vm machine. I don't know what should I do to do that, please help!

### Expected results

- A successfully built ipa file

### Actual results

- Receiving the following error at the 'Flutter build ipa' step:

Failed to build iOS app
Parse Issue (Xcode): Module 'connectivity_plus' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.

Build failed :|
Step 7 script `Flutter build ipa` exited with status code 1

### Build id (optional)

650261eb0bacb48f40d4e59f

**Comment by himesh-cm** [comment]
Hi @mohamed-gearbox, It Looks like a known issue with the plugin. Please refer to [this](https://github.com/fluttercommunity/plus_plugins/issues/1933) GitHub issue.

**Comment by mohamed-gearbox** [comment]
Thank you @himesh-cm , I carried out the steps per last comment on GitHub issue you referenced, however it didn't resolve the issue. After committing latest changes to repo and start a new app build, I'm still getting the same error:

Failed to build iOS app
Parse Issue (Xcode): Module 'connectivity_plus' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.

After removing the ios folder, then running "flutter create .", and running "flutter pub get", the following is the list of dependencies being resolved: (I don't see connectivity_plus in them...do I need to?), The app runs fine on android...

Resolving dependencies in ..... (1.8s)
  _fe_analyzer_shared 61.0.0 (64.0.0 available)
  _flutterfire_internals 1.3.1 (1.3.6 available)
  analyzer 5.13.0 (6.2.0 available)
  args 2.4.1 (2.4.2 available)
  async 2.10.0 (2.11.0 available)
  build 2.3.1 (2.4.1 available)
  build_daemon 3.1.1 (4.0.0 available)
  build_resolvers 2.2.0 (2.3.2 available)
  build_runner 2.3.3 (2.4.6 available)
  build_runner_core 7.2.2 (7.2.10 available)
  built_value 8.5.0 (8.6.2 available)
  characters 1.2.1 (1.3.0 available)
  checked_yaml 2.0.1 (2.0.3 available)
  code_builder 4.4.0 (4.6.0 available)
  collection 1.17.0 (1.18.0 available)
  cross_file 0.3.3+4 (0.3.3+5 available)
  cupertino_icons 1.0.5 (1.0.6 available)
  dart_openai 1.9.91 (4.0.0 available)
  dart_style 2.3.1 (2.3.3 available)
  dio 5.1.2 (5.3.2 available)
  elegant_notification 1.10.0 (1.11.1 available)
  fetch_client 1.0.1 (1.0.2 available)
  ffi 2.0.2 (2.1.0 available)
  file 6.1.4 (7.0.0 available)
  file_selector_linux 0.9.1+3 (0.9.2+1 available)
  file_selector_macos 0.9.2 (0.9.3+3 available)
  file_selector_platform_interface 2.5.1 (2.6.1 available)
  file_selector_windows 0.9.2 (0.9.3+1 available)
  firebase_auth 4.5.0 (4.10.0 available)
  firebase_auth_platform_interface 6.15.1 (6.19.0 available)
  firebase_auth_web 5.5.1 (5.8.1 available)
  firebase_core 2.13.0 (2.16.0 available)
  firebase_core_web 2.5.0 (2.8.0 available)
  flutter_cache_manager 3.3.0 (3.3.1 available)
  flutter_lints 2.0.1 (2.0.3 available)
  flutter_plugin_android_lifecycle 2.0.15 (2.0.16 available)
  flutter_secure_storage 6.1.0 (9.0.0 available)
  flutter_secure_storage_linux 1.1.3 (1.2.0 available)
  flutter_secure_storage_macos 1.1.2 (3.0.1 available)
  flutter_secure_storage_platform_interface 1.0.1 (1.0.2 available)
  flutter_secure_storage_web 1.1.1 (1.1.2 available)
  flutter_secure_storage_windows 1.1.3 (3.0.0 available)
  glob 2.1.1 (2.1.2 available)
  graphs 2.2.0 (2.3.1 available)
  http 0.13.6 (1.1.0 available)
  image_picker 0.8.9 (1.0.4 available)
  image_picker_android 0.8.7 (0.8.8 available)
  image_picker_for_web 2.2.0 (3.0.1 available)
  image_picker_ios 0.8.8 (0.8.8+2 available)
  image_picker_linux 0.2.1 (0.2.1+1 available)
  image_picker_macos 0.2.1 (0.2.1+1 available)
  image_picker_platform_interface 2.8.0 (2.9.1 available)
  image_picker_windows 0.2.1 (0.2.1+1 available)
  intl 0.17.0 (0.18.1 available)
  js 0.6.5 (0.6.7 available)
  json_serializable 6.6.2 (6.7.1 available)
  lints 2.0.1 (2.1.1 available)
  logging 1.1.1 (1.2.0 available)
  matcher 0.12.13 (0.12.16 available)
  material_color_utilities 0.2.0 (0.8.0 available)
  meta 1.8.0 (1.10.0 available)
  path 1.8.2 (1.8.3 available)
  path_provider 2.0.15 (2.1.1 available)
  path_provider_android 2.0.27 (2.2.0 available)
  path_provider_foundation 2.2.3 (2.3.1 available)
  path_provider_linux 2.1.11 (2.2.1 available)
  path_provider_platform_interface 2.0.6 (2.1.1 available)
  path_provider_windows 2.1.7 (2.2.1 available)
  permission_handler 10.4.3 (11.0.0 available)
  permission_handler_android 10.3.2 (11.0.1 available)
  permission_handler_platform_interface 3.11.3 (3.11.5 available)
  petitparser 5.1.0 (6.0.1 available)
  platform 3.1.0 (3.1.2 available)
  plugin_platform_interface 2.1.4 (2.1.6 available)
  process 4.2.4 (5.0.0 available)
  pubspec_parse 1.1.0 (1.2.3 available)
  retrofit_generator 5.0.0+1 (7.0.8 available)
  shimmer 2.0.0 (3.0.0 available)
  source_gen 1.3.2 (1.4.0 available)
  source_helper 1.3.3 (1.3.4 available)
  source_span 1.9.1 (1.10.0 available)
  sqflite 2.2.8+4 (2.3.0 available)
  sqflite_common 2.4.5+1 (2.5.0 available)
  stack_trace 1.11.0 (1.11.1 available)
  stream_channel 2.1.1 (2.1.2 available)
  syncfusion_flutter_charts 20.4.54 (23.1.36 available)
  syncfusion_flutter_core 20.4.54 (23.1.36 available)
  test_api 0.4.16 (0.6.1 available)
  timing 1.0.0 (1.0.1 available)
  tuple 2.0.1 (2.0.2 available)
  typed_data 1.3.1 (1.3.2 available)
  url_launcher 6.1.11 (6.1.14 available)
  url_launcher_android 6.0.35 (6.1.0 available)
  url_launcher_ios 6.1.4 (6.1.5 available)
  url_launcher_linux 3.0.5 (3.0.6 available)
  url_launcher_macos 3.0.5 (3.0.7 available)
  url_launcher_platform_interface 2.1.2 (2.1.5 available)
  url_launcher_web 2.0.17 (2.0.20 available)
  url_launcher_windows 3.0.6 (3.0.8 available)
  uuid 3.0.7 (4.0.0 available)
  watcher 1.0.2 (1.1.0 available)
  win32 4.1.4 (5.0.8 available)
  xdg_directories 1.0.0 (1.0.3 available)
  xml 6.2.2 (6.4.2 available)
Got dependencies in ..!

The error point to file -> /Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

When I checked the code part referenced by line number 11:
`#if __has_include(<connectivity_plus/ConnectivityPlusPlugin.h>)
#import <connectivity_plus/ConnectivityPlusPlugin.h>
#else
@import connectivity_plus;
#endif`

It seems to be failing at the import statement. Do I need to SSH into the instance and import this module to XCode running on m1 VM instance?

**Comment by mohamed-gearbox** [comment]
I've commented out reference / use of connectivity_plus module in the flutter project, just to see build on codemagic progress. This resolved the error about connectivity_plus not found, however now I'm getting...

Failed to build iOS app
Parse Issue (Xcode): Module 'firebase_auth' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.


Build failed :|
Step 7 script `Flutter build ipa` exited with status code 1

I hope someone could shed some light on this, if this could be resolved somehow by SSH into VM instance to carry out in configuration!

**Reply by himesh-cm** [reply]
Are you seeing the same error locally?

**Comment by mohamed-gearbox** [comment]
I'm running locally on Windows and VSCode (I have no access to an iMac or Macbook Pro to run ios app locally over XCode - hence using codemagic for this reason). 

That said, the app is running fine on android emulator and building app bundle aab release file is fine and deployed on Google Play and tried out successfully on android devices.

Note: I have included my codemagic.yaml configuration above, is there anything wrong that I made in this file that could be causing this?

**Reply by himesh-cm** [reply]
There is a configuration-related issue. Ideally, in some cases, you will need Xcode for local development.
You can check for solutions [online](https://www.google.co.in/search?q=module+%27firebase_auth%27+not+found+flutter+ios&sca_esv=568138803&source=hp&ei=F0wRZfm8Eua02roP8q-a4AU&iflsig=AO6bgOgAAAAAZRFaJ_zL8Vg-tsVksZonXORHWohqIoel&oq=Module+%27firebase_auth%27+not+found+flutter&gs_lp=Egdnd3Mtd2l6IihNb2R1bGUgJ2ZpcmViYXNlX2F1dGgnIG5vdCBmb3VuZCBmbHV0dGVyKgIIADIFEAAYgAQyCBAAGIoFGIYDMggQABiKBRiGA0iBE1AAWKAJcAB4AJABAJgBmgGgAfAJqgEDMC45uAEDyAEA-AEC-AEBwgIGEAAYFhge&sclient=gws-wiz#ip=1), and see if there are any workarounds.

**Comment by InnoquestInc** [comment]
I had exactly the same problem and solved with:

Removed any manually created podfile or podfile.lock files from my ios folder on my VScode project.
Then added this line: rm -rf ios; flutter create /Users/builder/clone/
to the pre-build script area in the codemagic build setup screen.  Finally I can succesfuly build my ios app with codemagic on my PC!

**Comment by mohamed-gearbox** [comment]
Thank you @InnoquestInc, I believe that I'm nearing a solution to my problem.

When using:
`rm -rf ios; flutter create /Users/builder/clone/`
or
`rm -rf ios; flutter create . --platforms ios`

During the flutter build ipa step it's now using com.example.clone as the bundle id, while my actual bundle identifier is com.example.ios and so it is failing with:

Archiving com.example.clone...
Automatically signing iOS for device deployment using specified development team in Xcode project: XXXXXXXXXX
Running pod install...                                           1,084ms
Running Xcode build...                                          
Xcode archive done.                                          2.4s
Failed to build iOS app
Error (Xcode): No profiles for 'com.example.clone' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.example.clone'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/ios/Runner.xcodeproj


It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.


Build failed :|
Step 8 script `Flutter build ipa` exited with status code 1

**Reply by himesh-cm** [reply]
Have you created developer profiles in the Apple developer account with bundle id `com.example.clone`?
Also, it is recommended to use an actual bundle identifier and not a dummy id like `com.example.clone`.

**Comment by mohamed-gearbox** [comment]
I'm just using example.com as an example, as it is industry standard when
sharing test/debug info.

Let's just say my developer profile uses a bundle identifier on Apple
Developer Account e.g. app.mydomain.ios however when  the build runs it
references app.mydomain.clone instead. I'm assuming it has to do with
recreating the flutter ios folder inside the /Users/builder/clone as a
build script step.

**Reply by himesh-cm** [reply]
If your bundle identifier is `com.example.clone` you need to create relevant profiles on the apple developer account using the same bundle id `com.example.clone`
You can check [this](https://docs.codemagic.io/yaml-code-signing/signing-ios/) guide for more details.

**Reply by mohamed-gearbox** [reply]
In codemagic.yaml the correct bundle identifier is defined i.e. app.mydomain.ios, this is the same bundle identifier on Apple Developer Account.

And so my question is why is it referencing app.mydomain.clone? Where is this string coming from, because it is not defined anywhere in my application code. It only popped up when using the following in the script section to create the ios folder and install podfile with dependencies correctly (which resolved the earlier problem of missing modules connectivity_plus, firebase_auth, etc...):

`rm -rf ios; flutter create /Users/builder/clone/`
or
`rm -rf ios; flutter create . --platforms ios`

**Reply by InnoquestInc** [reply, marked as answer]
The problem is when rebuild of ios files is done in script.  There is no name give in the command I first suggested and therefore xcode picks of the folder name "clone" as the bundle name.  I have fixed this problem with the following script command

`flutter create --project-name your_name_here .`

I also found that I needed to keep my icons in the ios folder so I removed the script to delete this folder.  Instead I just deleted the project.pbxproj file.  This is at ios>Runner.xcodeproj>project.pbxproj.  I found this to be the only item in the ios folder that was causing failure in the build and needing rebuild on CodeMagic via the above script.  Hope this helps... hours of time getting this to work for me!

**Reply by InnoquestInc** [reply]
Forgot one important thing > I had to move the script from the Pre-Build Script area up to Post-clone script for proper error free build

**Reply by mohamed-gearbox** [reply]
It finally worked and built the ipa file! Thank you so much @InnoquestInc 

Here is the script that worked for me
rm -f ios/Runner.xcodeproj/project.pbxproj; flutter create --project-name ios --org app.mydomain .

**Reply by himesh-cm** [reply]
Thanks, @InnoquestInc, for pitching in.
@mohamed-gearbox, While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!

**Reply by mohamed-gearbox** [reply]
Hello @InnoquestInc, @himesh-cm 

Given the past problem/solution, and as an addendum, is there a way to add iOS permissions (i.e. 'PERMISSION_CAMERA=1' and 'PERMISSION_PHOTOS=1') to the config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] in the podfile which is generated by the pre-build script?

I've added the following permissions in the Info.plist file:

```
<key>NSPhotoLibraryUsageDescription</key>
<string>This app uses the photo gallery to select photos</string>

<key>NSCameraUsageDescription</key>
<string>This app uses the camera to capture photos and videos</string>
```

however, the user is not being prompted to grant permissions (the alert dialog) on iOS devices, I believe the problem might be due to not adding these permissions in the podfile as well!? Is it imperative that these permissions be defined in the podfile, besides the Info.plist, or just having them in the Info.plist file should suffice?

**Reply by mohamed-gearbox** [reply]
I found a solution that worked, it's not the most elegant however did the trick for the short term!

In the spirit of sharing, here is the solution:

First and in the case of iOS, it is imperative to add any permissions needed in the Podfile along with it's relative keys in Info.plist as well.

Since the Podfile is created through the codemagic build process, as mentioned above, the only way I thought to do that was to execute a shell script that would inject the enabled permissions into the Podfile directly (in my case I specifically needed - 'PERMISSION_CAMERA=1', 'PERMISSION_PHOTOS=1',)

The following is the shell script I wrote, that was part of the project and gets executed from the codmagic.yaml file as a pre-build step.

```
#!/usr/bin/env bash

# Locate the podfile using find
PODFILE_PATH=ios/Podfile

# Check if the file exists
if [[ ! -f "$PODFILE_PATH" ]]; then
    echo "Error: Podfile not found."
    exit 1
fi

# Check for 'config.build_settings['GCC_PREPROCESSOR_DEFINITIONS']'
if grep -q "config.build_settings\['GCC_PREPROCESSOR_DEFINITIONS'\]" "$PODFILE_PATH"; then
    sed -i.bak '/config.build_settings\['\''GCC_PREPROCESSOR_DEFINITIONS'\''\].*\[/a \
    '\''PERMISSION_CAMERA=1'\'', '\''PERMISSION_PHOTOS=1'\'',\
    ' "$PODFILE_PATH"
elif grep -q "target.build_configurations.each do |config|" "$PODFILE_PATH"; then
    sed -i.bak '/target.build_configurations.each do |config|/a \
    \ \ config.build_settings\['\''GCC_PREPROCESSOR_DEFINITIONS'\''\] \|\|= \[ '\''$(inherited)'\'', '\''PERMISSION_CAMERA=1'\'', '\''PERMISSION_PHOTOS=1'\'', \]\
    ' "$PODFILE_PATH"
elif grep -q "installer.pods_project.targets.each do |target|" "$PODFILE_PATH"; then
    sed -i.bak '/installer.pods_project.targets.each do |target|/a \
	\ \ flutter_additional_ios_build_settings(target)  \
    \ \ target.build_configurations.each do |config|     config.build_settings\['\''GCC_PREPROCESSOR_DEFINITIONS'\''\] \|\|= \[ '\''$(inherited)'\'', '\''PERMISSION_CAMERA=1'\'', '\''PERMISSION_PHOTOS=1'\'', \] end\
    ' "$PODFILE_PATH"
elif grep -q "post_install do |installer|" "$PODFILE_PATH"; then
    sed -i.bak '/post_install do |installer|/a \
    installer.pods_project.targets.each do |target| flutter_additional_ios_build_settings(target) target.build_configurations.each do |config| config.build_settings\['\''GCC_PREPROCESSOR_DEFINITIONS'\''\] \|\|= \[ '\''$(inherited)'\'', '\''PERMISSION_CAMERA=1'\'', '\''PERMISSION_PHOTOS=1'\'', \] end end\
    ' "$PODFILE_PATH"
else
    echo "Did not find a suitable block to insert permissions. You might need to manually add them."
fi
```

I'd be interested to learn If there is a better more elegant way of accomplishing this!
